app/controllers/public_locations_controller.rb

```diff
@@ -0,0 +1,31 @@
+class PublicLocationsController < ApplicationController
+  before_action :set_location, only: :show
+  before_action :set_business, only: :index
+
+  def show
+    @business = @location.business
+  end
+
+  def index
+    @locations = @business.locations
+    if @locations.count < 2
```

Comment:
this should be `== 1`.  It's more obvious what you're doing when there's only one.  If there is zero, you might want to display an error page.